tools/verify-commit/verify-commit.go

```diff
@@ -94,4 +93,5 @@ func main() {
 			os.Exit(2)
 		}
 	}
+	fmt.Printf("done verifying commits from stdin\n")
```

Comment:
why this one?

Reply:
because on fatal error the function will exit immediately and not execute deferred code. 
Maybe a better way is actually replacing `Fatal` with `klog.Errorf(); return;` 
it is common that in many unit tests we use `fatal` and in others just `t.Error`. I prefer to go with `logging an error and return (or continue in case of a loop) because on a fatal error the code will completely exit as soon as it hit it and the rest of the tests won't execute.

Reply:
>because on fatal error the function will exit immediately and not execute deferred code.

Reply:
I prefer to move to `return X` and drop the explicit `os.Exit` calls rather than moving away from `defer`. Could you please try replacing `os.Exit()`s with `return`s?

Reply:
yes indeed it is due to exit ultimately, so by logging a fatal, it prints out the error to a writer and then exits. 
this is main program, we can't return codes in main() (the proposal was eventually rejected). I will test this with only return, if that turns out to be fine we can mention the error code in the fmt.FprintF.

Reply:
ah right, `main` doesn't return int. Sigh.
feel free to move all the logic into another function and call it from `main`. Please add a comment referencing/summarizing this convo.